src/SpoolmanManager.cpp

```
@@ -1648,6 +1648,48 @@ int SpoolmanManager::findFilamentNoLock(int vendorId, const char* material,
 
+void SpoolmanManager::recordLinkResult(int32_t spoolId, bool ok, const char* uid) {
+    if (xSemaphoreTake(cacheMutex_, pdMS_TO_TICKS(100)) != pdTRUE) return;
+    lastLinkResult_.spoolId = spoolId;
+    lastLinkResult_.ok = ok;
+    lastLinkResult_.atMs = millis();
+    strncpy(lastLinkResult_.uid, uid ? uid : "", sizeof(lastLinkResult_.uid) - 1);
+    lastLinkResult_.uid[sizeof(lastLinkResult_.uid) - 1] = '\0';
+    xSemaphoreGive(cacheMutex_);
+}
+
+SpoolmanManager::PendingLinkStatus SpoolmanManager::getPendingLinkStatus() {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)